src/components/search/search.tsx

```diff
@@ -1,44 +1,29 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import classes from './search.module.scss';
 
-interface ISearch {
-  textSearch: string;
-}
+const Search = () => {
+  const [textSearch, setSearch] = useState(localStorage.getItem('LocalStorageSearch') || '');
```

Comment:
for such calls better use callbacks:
```
const [textSearch, setSearch] = useState(() => localStorage.getItem('LocalStorageSearch') || '');
```

in ideal world such things should be done in useEffect hook)